### PR TITLE
I708 iiif image auth

### PR DESCRIPTION
**Pull request by maxkadel**
### From original PR:
This commit adds a new route check-iiif which tests for authorization in whatever way other items test for authorization.
It also adds Nginx configs to allow the Cantaloupe service to be moved to authorized-iiif and for this app to server /iiif.

All upstream image transfer after authorization would then be streamed by nginx and would free the Rails application process.

See https://medium.com/@jefawks3/using-nginx-as-a-multi-step-reverse-proxy-to-authorize-requests-before-serving-content-on-s3-d4094577c60e

Fixes https://github.com/yalelibrary/YUL-DC/issues/708

### New comment:
Reverted previous PR since it was not deployable (yet). 

image

https://github.com/yalelibrary/yul-dc-blacklight/pull/419

### CODE REVIEW

File: ops/webapp.conf [hunk 1]
```diff
@@ -24,4 +32,138 @@ server {
...
+
+        ## Adding a header to verify the request is coming from NGINX on the server
+        ##
+        ## TODO Find a better way not to store this in the file
+        ##  NGINX does not allow ENV variables, so you can use a
+        ##   deployment script ot update the value
+        ## proxy_set_header   X-Nginx-Secret   'some-super-secret-key';
+
+        proxy_redirect off; # We don't want a redirect to be passed to the client - NGINX needs to handle it
+        proxy_pass http://127.0.0.1:3000/check-iiif; # Send the user access request to the API server
```
Comment (mlooney) [inline ops/webapp.conf:86]: backend doesn't live on localhost anymore. $cluster_name.images.$cluster_name should do it

**Review by mlooney — CHANGES_REQUESTED** [review]

File: ops/webapp.conf (hunk 1 above)
Comment (mlooney) [reply ops/webapp.conf:86]: oops, it's actually thus: $cluster_name-images.$cluster_name

Comment (orangewolf) [reply ops/webapp.conf:86]: not even within the container? this is only an internal call (nginx is running in this container)

Comment (maxkadel) [reply ops/webapp.conf:86]: @orangewolf  - Did you see the comment on the revert? Tried deploying this code and got message below, and the app wouldn't start up at all. I'll try deploying this branch to test today and see if this works. I'm not sure I understand the proxying yet, though, so I'm not sure how successful I'll be.

image

**Review by mlooney — APPROVED** [review]

File: ops/webapp.conf (hunk 1 above)
Comment (orangewolf) [reply ops/webapp.conf:86]: sure max, but that isn't related to this line, its related to the one below that you changed =-)

**Comment by maxkadel** [comment]
Closing in favor of resolved PR - https://github.com/yalelibrary/yul-dc-blacklight/pull/425

**Pull request closed without merging**